src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReferenceImpl.kt

```diff
@@ -27,60 +29,62 @@ class RsPathReferenceImpl(
 
     override val RsPath.referenceAnchor: PsiElement get() = referenceNameElement
 
-    override fun resolveInner(): List<BoundElement<RsCompositeElement>> {
-        val lookup = ImplLookup.relativeTo(element)
-        val result = collectResolveVariants(element.referenceName) {
-            processPathResolveVariants(lookup, element, false, it)
-        }
+    override fun resolveInner(): List<BoundElement<RsCompositeElement>> =
+        (element.parent as? RsPathExpr)?.let { it.inference?.getResolvedPath(it) } ?: resolvePath(element)
```

Comment:
We can't (and don't need to) delegate all path resolution to type inference, but it seems not clear whether path depends on type inference or not. May be create a special reference class for path of `RsPathExpr`?..